addons/website/controllers/main.py

```diff
@@ -1046,6 +1046,35 @@ def actions_server(self, path_or_xml_id_or_id, **post):
 
         return request.redirect('/')
 
+    # Enable the footer template and its corresponding copyright template.
+    # The goal is to ensure that the content width of the copyright aligns with the footer.
+    @http.route(['/website/update_footer_template'], type='jsonrpc', auth='user', website=True)
+    def update_footer_template(self, template_key, possible_values):
+
+        # 1 . Get the content width of the new footer template.
+        new_template = self._get_customize_data([template_key], is_view_data=True)
+        if not new_template or not new_template[0].arch:
+            return
+        tree = etree.HTML(new_template[0].arch)
+        classes = tree.xpath(f"//div[contains(@class, 'container')]")[0].get('class').split()
```

Comment:
Here you assume the `xpath` query will always return at least one `<div>` containing valid classes. Try to make this more robust. You'll probably find a case where this will raise an IndexError (`[0]` access) or an AttributeError (`split`)
Furthermore, you're using an `fstring` without placeholder, that's not relevant.